packages/base/src/widget.ts

```diff
@@ -689,9 +689,9 @@ class WidgetView extends NativeView<WidgetModel> {
     /**
      * Create and promise that resolves to a child view of a given model
      */
-    create_child_view(child_model: WidgetModel, options = {}): Promise<DOMWidgetView> {
+    create_child_view<VT extends DOMWidgetView = DOMWidgetView>(child_model: WidgetModel, options = {}): Promise<VT> {
         options = { parent: this, ...options};
-        return this.model.widget_manager.create_view(child_model, options)
+        return this.model.widget_manager.create_view<VT>(child_model, options)
```

Comment:
I think once you have the overrides, you won't need to specify the type parameter here, as it should select the right override.

Reply:
Yes! Thanks you

Reply:
Actually it does not seem to compile without it:

```
src/widget.ts:696:9 - error TS2322: Type 'Promise<DOMWidgetView>' is not assignable to type 'Promise<VT>'.
  Type 'DOMWidgetView' is not assignable to type 'VT'.
    'DOMWidgetView' is assignable to the constraint of type 'VT', but 'VT' could be instantiated with a different subtype of constraint 'WidgetView'.

696         return this.model.widget_manager.create_view(child_model, options)
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
697             .catch(utils.reject('Could not create child view', true));
```

I feel like this compilation error is not legit though, as `DOMWidgetView` is a subclass of `WidgetView`, what do you think?